File: Project1/11047.cpp
#include <iostream>
#include <vector>

using namespace std;
int main(void)
{
    int N, K;
    cin >> N >> K;
    vector<int> coins(N);
    for (int i=0; i<N; ++i)
    {
        int val;
        cin >> val;
        coins[i] = val;
    }
    int result = 0;
    for (int i = N-1; i >= 0 && K != 0; --i)
    {
        if (K / coins[i] == 0)
        {
            continue;
        }
        result += K / coins[i];
        K = K % coins[i];
    }
    cout << result;
}

File: Project1/11047.py
import sys
input = sys.stdin.readline
n, k = map(int, input().split())
coins = []
for i in range(n):
    coins.append(int(input().strip()))
result = 0
for i in range(n-1, -1, -1):
    result += k // coins[i]
    k %= coins[i]
print(result)
